### ISSUE TITLE
Nuvoton I2C speed settings inconsistent with other devices

### ISSUE THREAD

**Issue by lightelf76** [OpenNuvoton/NUC970_U-Boot_v2016.11#8]
NUC970 driver measure I2C speed in kHz, where most other SoC uses value in Hz.

**Comment by yachen** [comment]
Yup, you're right. Do you want to send a pull request? Or we can update the code by our engineer.


Sincerely,

Yi-An Chen

**Comment by lightelf76** [comment]
I am thinking how to do it without disturbing existing customers. By the way, I started N32926 support for one of old projects.

**Comment by yachen** [comment]
Hi,

NUC970 TRM states “Only 100kbps and 400kbps modes are supported directly in this chip.”
So seeing anything below 1000, the driver can assume the unit is kHz, otherwise it’s Hz.
This should minimize the impact to existing customers those are using I2C function in U-Boot.

Sincerely,

Yi-An Chen


===========================================================================================
The privileged confidential information contained in this email is intended for use only by the addressees as indicated by the original sender of this email. If you are not the addressee indicated in this email or are not responsible for delivery of the email to such a person, please kindly reply to the sender indicating this fact and delete all copies of it from your computer and network server immediately. Your cooperation is highly appreciated. It is advised that any unauthorized use of confidential information of Nuvoton is strictly prohibited; and any information in this email irrelevant to the official business of Nuvoton shall be deemed as neither given nor endorsed by Nuvoton.

**Comment by lightelf76** [comment]
Done! Please see pull request
